MVVM-C/Controllers/Article Details/DetailViewVC.swift

```diff
@@ -0,0 +1,166 @@
+        view.addSubview(imageArticle)
+        view.addSubview(safariButton)
+        
+        
+        NSLayoutConstraint.activate([
+            articleLabel.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 20),
+            articleLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            articleLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            articleLabel.heightAnchor.constraint(equalToConstant: 40),
+            
+            abstractLabel.topAnchor.constraint(equalTo: articleLabel.bottomAnchor, constant: 10),
+            abstractLabel.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            abstractLabel.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            abstractLabel.heightAnchor.constraint(equalToConstant: 100),
+            
+            imageArticle.topAnchor.constraint(equalTo: abstractLabel.bottomAnchor, constant: 10),
+            imageArticle.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            imageArticle.widthAnchor.constraint(equalToConstant: 300),
+            imageArticle.heightAnchor.constraint(equalToConstant: 300),
+            
+            safariButton.topAnchor.constraint(equalTo: imageArticle.bottomAnchor, constant: 40),
+            safariButton.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            safariButton.heightAnchor.constraint(equalToConstant: 50),
+            safariButton.widthAnchor.constraint(equalToConstant: 250)
+        ])
+    }
+}
+
+
+class GoTLabel: UILabel {
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configure()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    convenience init(textAlignment: NSTextAlignment, fontSize: CGFloat) {
+        self.init(frame: .zero)
+        self.textAlignment = textAlignment
+        self.font = UIFont.systemFont(ofSize: fontSize, weight: .bold)
+    }
+    
+    func configure() {
+        textColor = .label
+        translatesAutoresizingMaskIntoConstraints = false
+        lineBreakMode = .byTruncatingTail
+        adjustsFontForContentSizeCategory = true
+        numberOfLines = 4
+    }
+}
+
+
+class GoTImageThumbnail: UIImageView {
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        configure()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func configure() {
+        layer.cornerRadius = 20
+        clipsToBounds = true
+        translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    func downloadThumbnail(_ link: String) {
+        NetworkManager.shared.downloadImage(from: link) { [weak self] (image) in
+            DispatchQueue.main.async {
+                self?.image = image
+            }
```

Comment:
do you think that this function should be in UIImageView class ? Do you think its ok ?

Reply:
I don't think it should be there, because if you want to change how the image is loaded or to test it, you could have problems. The best way is to set a Controller class to manage that image load for a UI object.